docs/developer-guide/cstor-pool-docs.md

```diff
@@ -0,0 +1,189 @@
+                            X                X  Reconcile     |                         |
+     XXX                   X                 +--------------->+                         |
+    XXXXX                  X                 X                |                         |
+      X                    X   CSPC-YAML     X                |                         |
+    XXXXX                  X                 X                |                         |
+      X        Applies     X                 X                +-------------------------+
+      X     +------------->X                 X
+     XXX                   X                 X
+    X X X                  X                 X
+      X                    X                 X
+                           X                 X
+                           XXXXXXXXXXXXXXXXXXX
+
+```
+
+1. CSPC contains a list of pool specs and a pool spec contains configuration for cStor pool creation on a particular node as specified in the pool spec.
+For example, if a CSPC manifest has 3 pool specs, 3 cStor pools will be created on 3 different nodes.
+
+2. An admin/SRE can preapre their CSPC YAML according to the reqiremnts and do a `kubectl apply -f <cspc-yaml-file>` to provision a CStorPoolCluster.
+
+3. Once a CSPC is applied, it is validated by cStor admission server and if the validation passes it is admitted to the etcd.
+
+4. The `CSPC-operator` receives events from Kubernetes APIs  for the created CSPC as well as existing CSPC and `CSPC-Operator` converges the system towards intent specified in the CSPC. Just a note that `CSPC-Operator` is deployed as a deployment controller and runs as a pod in the Kubernetes cluster.
+
+### What does CSPC-Operator do?
+
+1. CSPC Operator creates the following resources as part of its job to converge the system towards the intent specified in the CSPC:
+- CStorPoolInstance(CSPI) - CSPI is a Kubernetes custom resource like the CSPC.
+- Pool Manager Deployment - Deployment is a native Kubernetes resource.
+- BlockDeviceClaim(BDC) - BDC is again a custom resource. 
+
+2. Let us say, CSPC has `k` pool specs then exactly `k` CSPI and `k` pool manager deployment is created. Each CSPI created corresponds to only one pool manger. So we can say that CSPI and pool manager has a one to one mapping.
+
+2. Each pool spec specified in the CSPC is converted into a CSPI resource and created to persist in etcd. For each CSPI a corresponding pool manager is also created. 
+
+3. For each block device mentioned in the CSPC YAML for all the pool specs, CSPC-Operator creates a BDC to ensure exclusive ownership over the disks. As discussed earlier, disks are represented as BlockDevice(BD) custom resource. Hence, there is a one to one mapping between BDC and BD.  
+
+3. The way CSPC is reconciled by CSPC-operator, in the same way a pool-manger reconciles its corresponding CSPI resource. It is important to note that a pool manager reconciles only one CSPI that corresponds to the pool manager.
```

Comment:
Q: Should we need to say **How pool manager will identify it's CSPI**?

Reply:
In this doc, I think we can skip it. Just to keep it simple. But we can have a detailed section on pool manager working and once we have that doc we can put a link here for more interested readers.